index.js

```diff
@@ -1,176 +1,179 @@
-var inherits = require('util').inherits
-var RandomAccess = require('random-access-storage')
-var fs = require('fs')
-var mkdirp = require('mkdirp-classic')
-var path = require('path')
-var constants = fs.constants || require('constants') // eslint-disable-line
-
-var READONLY = constants.O_RDONLY
-var READWRITE = constants.O_RDWR | constants.O_CREAT
-
-module.exports = RandomAccessFile
-
-function RandomAccessFile (filename, opts) {
-  if (!(this instanceof RandomAccessFile)) return new RandomAccessFile(filename, opts)
-  RandomAccess.call(this)
-
-  if (!opts) opts = {}
-  if (opts.directory) filename = path.join(opts.directory, path.resolve('/', filename).replace(/^\w+:\\/, ''))
-
-  this.directory = opts.directory || null
-  this.filename = filename
-  this.fd = 0
-
-  // makes random-access-storage open in writable mode first
-  if (opts.writable || opts.truncate) this.preferReadonly = false
-
-  this._size = opts.size || opts.length || 0
-  this._truncate = !!opts.truncate || this._size > 0
-  this._rmdir = !!opts.rmdir
-  this._lock = opts.lock || noLock
-  this._sparse = opts.sparse || noLock
-  this._alloc = opts.alloc || Buffer.allocUnsafe
-}
+const RandomAccessStorage = require('random-access-storage')
+const fs = require('fs')
+const mkdirp = require('mkdirp-classic')
+const path = require('path')
+const constants = fs.constants || require('constants') // eslint-disable-line n/no-deprecated-api
 
-inherits(RandomAccessFile, RandomAccess)
+const READONLY = constants.O_RDONLY
+const READWRITE = constants.O_RDWR | constants.O_CREAT
 
-RandomAccessFile.prototype._open = function (req) {
-  var self = this
+module.exports = function randomAccessFile (filename, opts) {
+  return new RandomAccessFile(filename, opts)
```

Comment:
does this pattern work if someone did new RandomAccessFile from the outside?

Reply:
Yep:

```
> const RAF = require('.')
undefined
> new RAF('foo.txt')
RandomAccessFile {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  _queued: [],
  _pending: 0,
  _needsOpen: true,
  opened: false,
  closed: false,
  destroyed: false,
  preferReadonly: true,
  readable: true,
  writable: true,
  deletable: true,
  statable: true,
  directory: null,
  filename: 'foo.txt',
  fd: 0,
  _size: 0,
  _truncate: false,
  _rmdir: false,
  _lock: [Function: noLock],
  _sparse: [Function: noLock],
  _alloc: [Function: allocUnsafe],
  [Symbol(kCapture)]: false
}
```

Reply:
+1 then, been thinking if we should just eat the complexity and move to this always being a class directly btw

Reply:
That does break the reasonably nice way of passing storage constructor functions as arguments:

```
const core = new Hypercore(RAF)
```

Reply:
or we simplify the api and have either all functions or all classes (and always call `new`)

```js
const core = new Hypercore(filename => new RAF(filename))
```

```js
const core = new Hypercore(RAF)
```

Reply:
After discussing it with @mafintosh, that's the direction we'll go with, yes. It felt like the approach with the fewest surprises.